### PR TITLE
fix(header): `search` should be always visible in PC UI

### PR DESCRIPTION
**Pull request by Daryl-L**
https://github.com/Magickbase/ckb-explorer-public-issues/issues/472
https://github.com/Magickbase/ckb-explorer-public-issues/issues/473

### CODE REVIEW

**Review by Keith-CY — APPROVED** [review]

**Comment by WhiteMinds** [comment]
The layout logic has changed, which is likely unexpected.

New:

image

Original:

image

File: src/utils/hook.ts [hunk 1]
```diff
@@ -367,7 +367,7 @@ export function useMediaQuery(query: string): boolean {
 }
 
 export const MOBILE_DEVICE_MAX_WIDTH = 750
-export const useIsMobile = () => useMediaQuery(`(max-width: ${MOBILE_DEVICE_MAX_WIDTH}px)`)
+export const useIsMobile = (max?: number) => useMediaQuery(`(max-width: ${max ?? MOBILE_DEVICE_MAX_WIDTH}px)`)
```
Comment (WhiteMinds) [inline src/utils/hook.ts:370]: `useIsMobile` is actually a JavaScript version of the 750px breakpoint. It should be a fixed value that is equal to the breakpoint defined in CSS, so it should not accept a variable parameter.

**Review by WhiteMinds — CHANGES_REQUESTED** [review]

**Comment by Daryl-L** [comment]
> The layout logic has changed, which is likely unexpected.
> 
> New:
> 
> image
> 
> Original:
> 
> image


This is the reason here, but why? I didn't find it in the design draft

**Comment by Keith-CY** [comment]
> The layout logic has changed, which is likely unexpected.
> 
> New:
> 
> image
> 
> Original:
> 
> image
> 
> Uploading image.png…
> This is the reason here, but why? I didn't find it in the design draft

Images are broken

**Comment by Daryl-L** [comment]
https://github.com/Magickbase/ckb-explorer-frontend/blob/db52e40231355f2b8e6db8fb109b0e6a306855de/src/index.css#L46

**Comment by Keith-CY** [comment]
> https://github.com/Magickbase/ckb-explorer-frontend/blob/db52e40231355f2b8e6db8fb109b0e6a306855de/src/index.css#L46

Which comment is this replied to

**Review by WhiteMinds — APPROVED** [review]

**Comment by Keith-CY** [comment]
Conflicted

**Comment by Keith-CY** [comment]
output

Notice the layout at 0:22 and 0:33
0:22, items in the header are connected to each other
0:33, search field in the dropdown is connected to the border, but a span is added when the width is smaller

**Comment by FrederLu** [comment]
The search box switching is weird.
Whether to switch to the top only when the lower search box is completely blocked, or to hide the top search box when the search box is about to appear on the page.

https://github.com/Magickbase/ckb-explorer-frontend/assets/25676576/9a8044f4-a190-4cc7-973f-3facd670a0f3

**Comment by Daryl-L** [comment]
> output [ output ](https://private-user-images.githubusercontent.com/7271329/292950478-9c9c8d5a-6e80-48a2-8841-246d7f0ddf7a.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQxNjkwNDksIm5iZiI6MTcwNDE2ODc0OSwicGF0aCI6Ii83MjcxMzI5LzI5Mjk1MDQ3OC05YzljOGQ1YS02ZTgwLTQ4YTItODg0MS0yNDZkN2YwZGRmN2EuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMDJUMDQxMjI5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDRhOWVlNmFhZTBhMjQxMjdkYmU0MzA4NWQwNDVhNmRlZTUxNjJjY2JlOTMxMzYwZWRmZTFmYzQ2MzI1ZDgzMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Ax_Ua2icjyBmX2f-fyIYD3XEoNPtD5bhU3TM8kNwTFA) [ ](https://private-user-images.githubusercontent.com/7271329/292950478-9c9c8d5a-6e80-48a2-8841-246d7f0ddf7a.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQxNjkwNDksIm5iZiI6MTcwNDE2ODc0OSwicGF0aCI6Ii83MjcxMzI5LzI5Mjk1MDQ3OC05YzljOGQ1YS02ZTgwLTQ4YTItODg0MS0yNDZkN2YwZGRmN2EuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMDJUMDQxMjI5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDRhOWVlNmFhZTBhMjQxMjdkYmU0MzA4NWQwNDVhNmRlZTUxNjJjY2JlOTMxMzYwZWRmZTFmYzQ2MzI1ZDgzMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Ax_Ua2icjyBmX2f-fyIYD3XEoNPtD5bhU3TM8kNwTFA)
> 
> Notice the layout at 0:22 and 0:33 0:22, items in the header are connected to each other 0:33, search field in the dropdown is connected to the border, but a span is added when the width is smaller

for 0:22, different from the design draft.
1. there is an extra item in the search component, but not found in desgin draft.
2. I found that the size between 1024px and 1920px is not listed in the design draft, so I just make them to auto adjust

**Comment by FrederLu** [comment]
image
image

1. Changing the resolution will also cause copywriting overlap.
2. At `1024px * 764px`, the search box will suddenly change size, but the default information in the search is gone, and the cursor will not appear when clicking the search box.
3. In Screenshot 2, it is found that the width of the top bar has changed.


---
2、The search box size on the mobile phone is abnormal.
imageimage

**Comment by FrederLu** [comment]
image
This problem still exists.

**Review by FrederLu — APPROVED** [review]

**Pull request merged**